src/views/signUpBuyer.jsx

```diff
@@ -0,0 +1,136 @@
+import React, { useState } from 'react';
+import { NavLink, useNavigate } from 'react-router-dom';
+import { ToastContainer, toast } from 'react-toastify';
+import CART from '../assets/CART.png';
+
+function SignUpBuyer() {
+  const navigate = useNavigate();
+  const [btn, setbtn] = useState({
+    display: 'hidden',
+    text: `Sign Up`,
+  });
+
+  const allFields = [
+    { placeh: 'enter first name ', label: ' first name', type: 'text' },
+    { placeh: 'enter last name ', label: ' last name', type: 'text' },
+    { placeh: 'enter email ', label: ' email', type: 'email' },
+    { placeh: 'phone ', label: ' phone', type: 'tel' },
+    { placeh: 'password ', label: ' password', type: 'password' },
+    {
+      placeh: 'confirm password ',
+      label: 'confirm password',
+      type: 'password',
+    },
+  ];
+  const SignUp = async (e) => {
+    e.preventDefault();
+    setbtn({
+      display: 'inline-block',
+      text: `Loading...`,
+    });
+    const phoneNumberRegex = /[a-zA-Z]/;
+    if (phoneNumberRegex.test(e.target[3].value) == true) {
```

Comment:
Extract this phone number validation code into a separate function. This would make the code more modular and easier to maintain.